helm search raises a runtime error unless helm has been initialized

After installing helm from source, if I execute `helm search redis` I get:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x40dfd0]

goroutine 1 [running]:
panic(0x1905e80, 0xc82000e120)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
main.searchCacheForPattern.func1(0xc8203df680, 0x26, 0x0, 0x0, 0x7f53d2f6f2d0, 0xc8203df800, 0x0, 0x0)
	/home/migmartri/work/gocode/src/github.com/kubernetes/helm/cmd/helm/search.go:63 +0x40
```

Problem that seems to be related to the fact that I have not initialized helm yet.

So executing `helm init` fixed the issue.

```
$ helm search redis
kubernetes-charts/redis-2.0.0.tgz
```

I have noticed that for some other commands there is a guard statement that tells you that the file does not exist, for example:

```
$helm fetch foo
Error: open /home/migmartri/.helm/repository/repositories.yaml: no such file or directory
```

We might want to implement the same approach for the search command. Ideally with a message a little bit more actionable that says something about executing helm init first (although this could be moved to another issue)

What do you think?

--- a/cmd/helm/helm.go
+++ b/cmd/helm/helm.go
@@ -109,6 +109,15 @@ func checkArgsLength(expectedNum, actualNum int, requiredArgs ...string) error {
 	return nil
 }
 
+// requireInit is a PreRunE implementation for validating that $HELM_HOME is configured.
+func requireInit(cmd *cobra.Command, args []string) error {
+	err := requireHome()
+	if err != nil {
+		return fmt.Errorf("%s (try running 'helm init')", err)
+	}
+	return nil
+}
+
 // prettyError unwraps or rewrites certain errors to make them more user-friendly.
 func prettyError(err error) error {
 	// This is ridiculous. Why is 'grpc.rpcError' not exported? The least they

--- a/cmd/helm/init.go
+++ b/cmd/helm/init.go
@@ -76,6 +76,19 @@ func installTiller() error {
 	return nil
 }
 
+// requireHome checks to see if $HELM_HOME exists, and returns an error if it does not.
+func requireHome() error {
+	dirs := []string{homePath(), repositoryDirectory(), cacheDirectory(), localRepoDirectory()}
+	for _, d := range dirs {
+		if fi, err := os.Stat(d); err != nil {
+			return fmt.Errorf("directory %q is not configured", d)
+		} else if !fi.IsDir() {
+			return fmt.Errorf("expected %q to be a directory", d)
+		}
+	}
+	return nil
+}
+
 // ensureHome checks to see if $HELM_HOME exists
 //
 // If $HELM_HOME does not exist, this function will create it.

--- a/cmd/helm/search.go
+++ b/cmd/helm/search.go
@@ -17,10 +17,11 @@ func init() {
 }
 
 var searchCmd = &cobra.Command{
-	Use:   "search [keyword]",
-	Short: "Search for a keyword in charts",
-	Long:  "Searches the known repositories cache files for the specified search string, looks at name and keywords",
-	RunE:  search,
+	Use:     "search [keyword]",
+	Short:   "Search for a keyword in charts",
+	Long:    "Searches the known repositories cache files for the specified search string, looks at name and keywords",
+	RunE:    search,
+	PreRunE: requireInit,
 }
 
 func search(cmd *cobra.Command, args []string) error {
@@ -45,6 +46,9 @@ func searchChartRefsForPattern(search string, chartRefs map[string]*repo.ChartRe
 	for k, c := range chartRefs {
 		if strings.Contains(c.Name, search) {
 			matches = append(matches, k)
+			continue
+		}
+		if c.Chartfile == nil {
 			continue
 		}
 		for _, keyword := range c.Chartfile.Keywords {

